internal/app/config.go

```diff
@@ -1,309 +1,121 @@
 package app
 
 import (
-	"net/url"
 	"os"
 	"strings"
-	"time"
 
-	"github.com/jeremywohl/flatten"
 	"github.com/metal-toolbox/flipflop/internal/model"
-	"github.com/mitchellh/mapstructure"
+	"github.com/metal-toolbox/flipflop/internal/store/fleetdb"
+	"github.com/metal-toolbox/rivets/events"
 	"github.com/pkg/errors"
-	"go.hollow.sh/toolbox/events"
 )
 
 var (
 	ErrConfig = errors.New("configuration error")
 )
 
-// Config holds application configuration read from a YAML or set by env variables.
-//
-// nolint:govet // prefer readability over field alignment optimization for this case.
+// Configuration values are first grabbed from the config file. They must be in the right place in order to be grabbed.
+// Values are then grabbed from the ENV variables, anything found will be used to override values in the config file.
+// Example: Setting Configuration.Endpoints.FleetDB.URL
+// In the config file (as yaml); endpoints.fleetdb.url: http://fleetdb:8000
+// As a ENV variable; FLIPFLOP_ENDPOINTS_FLEETDB_URL=http://fleetdb:8000
 type Configuration struct {
+	// FacilityCode limits this flipflop to events in a facility.
+	FacilityCode string `mapstructure:"facility"`
+
 	// LogLevel is the app verbose logging level.
 	// one of - info, debug, trace
 	LogLevel string `mapstructure:"log_level"`
 
-	// flipflop configuration
-	Concurrency int `mapstructure:"concurrency"`
+	StoreKind string `mapstructure:"store_kind"`
 
-	// FacilityCode limits this flipflop to events in a facility.
-	FacilityCode string `mapstructure:"facility_code"`
+	// Holds all endpoints
+	Endpoints Endpoints `mapstructure:"endpoints"`
 
-	// The inventory source - one of fleetdb OR Yaml
-	InventorySource string `mapstructure:"inventory_source"`
-
-	StoreKind model.StoreKind `mapstructure:"store_kind"`
+	// nats controller concurrency
+	Concurrency int `mapstructure:"concurrency"`
 
-	// FleetDBOptions defines the fleetdb client configuration parameters
-	//
-	// This parameter is required when StoreKind is set to fleetdb.
-	FleetDBOptions *FleetDBOptions `mapstructure:"fleetdb"`
+	// In dryrun mode, the worker actions the task without installing firmware
+	// Note: Currently completely overrided by commandline arg `--dry-run`
+	Dryrun bool `mapstructure:"dryrun"`
 
-	// EventsBrokerKind indicates the kind of event broker configuration to enable,
-	//
-	// Supported parameter value - nats
-	EventsBorkerKind string `mapstructure:"events_broker_kind"`
+	// Tasks can include a Fault attribute to allow fault injection for development purposes
+	// Note: Currently completely overrided by commandline arg `--fault-injection`
+	FaultInjection bool `mapstructure:"fault_injection"`
+}
 
+type Endpoints struct {
```

Comment:
if we're not referencing ConditionOrc here (or Condition API) should we remove that from the config file?

Reply:
Yes.